packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -618,7 +618,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         const restNode: N.TsRestType = this.startNode();
         this.next(); // skips ellipsis
         restNode.typeAnnotation = this.tsParseType();
-        this.checkCommaAfterRest(charCodes.rightSquareBracket);
+        if (
+          this.match(tt.comma) &&
+          this.lookaheadCharCode() !== charCodes.rightSquareBracket
```

Comment:
Is there already a test for when `...a,` is not followed by `]`?

Reply:
@nicolo-ribaudo 
 yes!  [tuple-rest-invalid-case](https://github.com/babel/babel/blob/e74efd2c0a03badef66c28f29d566967ac8af0ef/packages/babel-parser/test/fixtures/typescript/types/tuple-rest-invalid/input.ts#L1) in type tests